### DISCUSSION TITLE
How to Insert a middleware into the Ocelot pipeline?

### DISCUSSION THREAD

**Discussion by Ivony** [ThreeMammals/Ocelot#1898] [Q&A] [answered]
it's implement as a branch. we need insert a middleware into exists pipeline. it's useful.

https://github.com/ThreeMammals/Ocelot/blob/41d4f9c6dfa3069dc9035e2e82d0c46f7f08cf56/src/Ocelot/Middleware/Pipeline/OcelotPipelineBuilderExtensions.cs#L174

**Comment by raman-m** [comment, marked as answer]
Hi Ivony!
Just fork our repo, and develop! 😉 

Current [Middleware Injection](https://ocelot.readthedocs.io/en/latest/features/middlewareinjection.html) allows to override limited number of middlewares. Seems you want to rebuild the pipeline. It is possible in case of custom development in your fork!
But we are not going to add very hacky solutions of the pipeline because it will break current features with high probability.
